base/expr.jl

```diff
@@ -256,6 +256,23 @@ macro polly(ex)
     esc(isa(ex, Expr) ? pushmeta!(ex, :polly) : ex)
 end
 
+"""
+    @nonans
+
+Allows the compiler to assume no NaNs on floating-point operations.
+
+```julia
+@nonans function mathfunc(x)
+    #=
+        Function Definition
+    =#
+end
+```
+"""
+macro nonans(ex)
```

Comment:
This should probably be split out into a macro that takes the function and one that inserts the meta node. See `Base.@_noinline_meta` and `Base.@noinline`

Reply:
Not sure what you mean, from what I understand @noinline works the same as this, and @_noinline_meta only exists as an alternative to be used before this file is included in the standard library. Would it be really necessary?

Reply:
It is helpful to have the `_meta` version if you are using generated functions etc.

Reply:
Included this macro. Also made an LLVM IR test similar to noinline.